src/game/client/components/menus_browser.cpp

```diff
@@ -487,7 +487,10 @@ void CMenus::RenderServerbrowserServerList(CUIRect View)
 		if(Input()->MouseDoubleClick() && DoubleClicked)
 		{
 			if(Client()->State() == IClient::STATE_ONLINE && Client()->GetCurrentRaceTime() / 60 >= g_Config.m_ClConfirmDisconnectTime && g_Config.m_ClConfirmDisconnectTime >= 0)
-				m_Popup = POPUP_DISCONNECT;
+			{
+				m_Popup = POPUP_SWITCH_SERVER;
+				str_copy(m_aNextServer, g_Config.m_UiServerAddress, sizeof(g_Config.m_UiServerAddress));
```

Comment:
Oops, one more thing I haven't noticed on the first read, sorry. :(

Here and below, `str_copy`'s third parameter must be the size of the target buffer, i.e. it must be `sizeof(m_aNextServer)` here.
```suggestion
				str_copy(m_aNextServer, g_Config.m_UiServerAddress, sizeof(m_aNextServer));
```

Reply:
Yeah I used it the wrong way around the first time (wrote `m_aNextServer` into the `UiServerAddress`) and therefore forgot to switch the size too :O